Практика

nikitaSua opened the pull request:
Сделал linq задачи

nikitaSua committed:
Практика

Сделал linq задачи

DaniilManita commented on TaskInCl_2/Program.cs:
```diff
@@ -0,0 +1,114 @@
+                    new Customer(20, "Anna Ledesma", new DateTime(2017, 12, 29), 0.8)
+                };
+
+            //SearchMethods.FirstCustomer(customers);
+
+            //Console.WriteLine(SearchMethods.avarageBalance(customers));
+
+            //SearchMethods.SortByDateInRange(customers, new DateTime(2011, 7, 10), new DateTime(2017, 10, 2));
+            //SearchMethods.SortByDateInRangeV2(customers, new DateTime(2017, 10, 2), new DateTime(2011, 7, 10));
+
+            //SearchMethods.SearchByName(customers, "Margert");
+            //SearchMethods.SortFieldAndDirection<Customer>(customers, "Name");
```
что за комменты в коде ?

DaniilManita commented on TaskInCl_2/Program.cs:
```diff
@@ -0,0 +1,114 @@
+
+
+            } while (true);
+
+
+
+
+
+
+
+
+
+        }
```
что это такое ????

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+    {
+        public static void FirstCustomer(List<Customer> customers)
+        {
+            var minDate = customers.Min(c => c.RegistrationDate);
+            var firstUser = customers.Select(c=>c).Where(c=> c.RegistrationDate == minDate);
```
можно ли сделать сортировку по дате и потом сделать фертс ор дефолт ?

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+                        ($"Id:  {item.Id}, Name: {item.Name}, RegDate: {item.RegistrationDate}, Ballance: {item.Balance}");
+            }
+        }
+        public static void avarageBalance(List<Customer> customers)
```
нейминг

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+            var avarage = customers.Average(c => c.Balance);
+            Console.WriteLine(avarage);
+        }
+        public static void searchById(List<Customer> customers, int searchId)
```
нейминг

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+        {
+            bool flag = true;
+
+            if (startD.CompareTo(finishD) == 1)
+            {
+
+                DateTime temp = startD;
+                startD = finishD;
+                finishD = temp;
+                flag = false;
+            }
```
для чего тебе это ?

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+                flag = false;
+            }
+
+            var sortByStartDates = customers.Select(c => c).Where(c => c.RegistrationDate >= startD);
+            var sortByfinishDates = sortByStartDates.Select(c => c).Where(c => c.RegistrationDate <= finishD);
```
это можно сделать в одном запросе

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+                ordered = sortByfinishDates.OrderByDescending(c => c.RegistrationDate);
+            }
+
+            if (ordered.Count() != 0)
```
я говорил вам за каунт

DaniilManita commented on TaskInCl_2/SearchMethods.cs:
```diff
@@ -0,0 +1,145 @@
+
+        public static void SortFieldAndDirection<T>(List<Customer> customers, string property, bool direct)
+        {
+
```
что за отступы ?

nikitaSua committed:
Фикс замечаний к заданиям

Поправил Название Методов (теперь с большой)
Поменял все Count() на Any()
Объединил некоторые запросы в 1 
Убрал лишние отступы в коде и комментарии
